contracts/HubPool.sol

```diff
@@ -472,6 +498,43 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         );
     }
 
+    function _exchangeRateCurrent(address l1Token) internal returns (uint256) {
```

Comment:
this function is very similar to that implemented in v1. main differences are we now consider all variables on the associated `pooledToken` `l1Token` and I cleaned up/simplified a small bit of the logic.